### PR TITLE
Add non delta dbfs table migration (What.DBFS_ROOT_NON_DELTA) in migrate_table workflow

### LINKED ISSUE #332: [FEATURE]: Connect `_migrate_table_create_ctas` to `migrate-tables` workflow
most of the code is done in:
- https://github.com/databrickslabs/ucx/pull/1434

Diff:
https://github.com/databrickslabs/ucx/blob/9cb9dbf05ba26d6905a53ebbf964f25ce1bddc09/src/databricks/labs/ucx/hive_metastore/table_migrate.py#L286-L299

Tasks:
- [x] Copy this table using CTAS
- [ ] Copy the Metadata with `CREATE TABLE ... LIKE ...`
- [ ] Skip tables as needed based on size threshold or an exception list
- [x] Update the source table with `upgraded_to` property
- [ ] Link it to the workflow as task

### PR DESCRIPTION
**Pull request by HariGS-DB**
## Changes

There are 3 workflows for migrating tables.

migrate_tables: covers EXTERNAL_SYNC, DBFS_ROOT_DELTA, VIEW
migrate-external-hiveserde-tables-in-place-experimental: covers EXTERNAL_HIVESERDE and VIEW
migrate-external-tables-ctas: covers EXTERNAL_NO_SYNC, EXTERNAL_HIVESERDE and VIEW

 - This change adds the remaining scenario DBFS_ROOT_NON_DELTA to the  migrate_tables workflow
 - Also adds AclMigrationWhat.PRINCIPAL to the ACL migration strategy

Resolves #332 

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [X] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [ ] manually tested
- [X] added unit tests
- [X] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

File: src/databricks/labs/ucx/hive_metastore/workflows.py [hunk 1]
```diff
@@ -112,7 +133,9 @@ def migrate_views(self, ctx: RuntimeContext):
         """This workflow task migrates views from the Hive Metastore to the Unity Catalog using create view sql statement.
         It is dependent on the migration of the tables.
         """
-        ctx.tables_migrator.migrate_tables(what=What.VIEW, acl_strategy=[AclMigrationWhat.LEGACY_TACL])
+        ctx.tables_migrator.migrate_tables(
+            what=What.VIEW, acl_strategy=[AclMigrationWhat.LEGACY_TACL, AclMigrationWhat.PRINCIPAL]
```
Comment (nkvuong) [inline src/databricks/labs/ucx/hive_metastore/workflows.py:137]: nit - add a trailing comma so `make fmt` would align it nicely
```suggestion
            what=What.VIEW, acl_strategy=[AclMigrationWhat.LEGACY_TACL, AclMigrationWhat.PRINCIPAL],
```

File: src/databricks/labs/ucx/hive_metastore/workflows.py [hunk 2]
```diff
@@ -12,29 +12,48 @@ def __init__(self):
...
-        - For AWS: TBD
+        - For Azure or AWS: `principal-prefix-access`, `create-table-mapping`, `create-uber-principal`, `migrate-credentials`, `migrate-locations`, `create-catalogs-schemas`
+        """
+        ctx.tables_migrator.migrate_tables(
+            what=What.DBFS_ROOT_DELTA, acl_strategy=[AclMigrationWhat.LEGACY_TACL, AclMigrationWhat.PRINCIPAL]
+        )
+
+    @job_task(job_cluster="table_migration", depends_on=[Assessment.crawl_tables])
+    def migrate_dbfs_root_non_delta_tables(self, ctx: RuntimeContext):
+        """This workflow task migrates `non delta tables stored in DBFS root` from the Hive Metastore to the Unity Catalog using CTAS.
```
Comment (nkvuong) [inline src/databricks/labs/ucx/hive_metastore/workflows.py:33]: could you standardise the formatting of these docstrings? we currently have 3 different styles
- This workflow task migrates the *external tables that are supported by SYNC command* 
- This workflow task migrates `delta tables stored in DBFS root`
- This workflow task migrates views from the Hive Metastore

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -783,6 +783,9 @@ def prepare_hiveserde_tables(context, random, schema, table_base_dir) -> dict[st
 def prepare_regular_tables(context, external_csv, schema) -> dict[str, TableInfo]:
     tables: dict[str, TableInfo] = {
         "src_managed_table": context.make_table(schema_name=schema.name),
+        "src_managed_non_delta_table": context.make_table(
```
Comment (nkvuong) [inline tests/integration/conftest.py:786]: do we need to change anything else to verify that integration tests would work correctly?

File: src/databricks/labs/ucx/hive_metastore/workflows.py (hunk 1 above)
Comment (HariGS-DB) [reply src/databricks/labs/ucx/hive_metastore/workflows.py]: done

File: src/databricks/labs/ucx/hive_metastore/workflows.py (hunk 2 above)
Comment (HariGS-DB) [reply src/databricks/labs/ucx/hive_metastore/workflows.py]: updated formatting. Also removed the requirement for pre-requisite cli cmd, it was repetitive and also mentioned in the read me doc

File: tests/integration/conftest.py (hunk 3 above)
Comment (HariGS-DB) [reply tests/integration/conftest.py:786]: No. I check this additional was covering the scenario for non delta dbfs table. screenshot below
table created
image

screenshot from migrate non delta dbfs workflow task log file
image

Comment (HariGS-DB) [reply tests/integration/conftest.py:786]: Due to the change in https://github.com/databrickslabs/ucx/pull/1623, i had to add the following code in the test_table_migration_job_refreshes_migration_status as it was not updating the migration_status for the last view migrated.
ctx.migration_status_refresher.reset()
    ctx.migration_status_refresher.snapshot()

Comment (nkvuong) [reply tests/integration/conftest.py:786]: I've fixed this in a separate PR #1630

File: tests/integration/hive_metastore/test_workflows.py [hunk 4]
```diff
@@ -26,6 +26,8 @@ def test_table_migration_job_refreshes_migration_status(ws, installation_ctx, pr
 
     ctx.workspace_installation.run()
     ctx.deployed_workflows.run_workflow(workflow)
+    ctx.migration_status_refresher.reset()
+    ctx.migration_status_refresher.snapshot()
```
Comment (nkvuong) [inline tests/integration/hive_metastore/test_workflows.py:30]: we don't need this fix anymore
```suggestion
```

**Review by nkvuong — APPROVED** [review]

**Review by nfx — APPROVED** [review]
Lgtm

**Pull request merged**